lib/connectors/memory.js

```diff
@@ -421,6 +421,11 @@ function applyFilter(filter) {
       // Also support array types. Mongo, possibly PostgreSQL
       if (Array.isArray(value)) {
         var matcher  = where[key];
+        // The following condition is for the case where we are querying with
+        // a neq filter, and when the value is an empty array ([]).
+        if (matcher.neq !== undefined && value.length <= 0) {
```

Comment:
I made the test, and {neq: [1, 2]} does not work correctly. The code written in the commit 632898b clearly doesn't handle this kind of cases since it makes a one by one analysis of the 'value' array.